### ISSUE TITLE
Differential evaluation crossover_gap = 0

### ISSUE THREAD

**Issue by dklong-csu** [bangerth/SampleFlow#181]
If `crossover_gap` in `DifferentialEvaluationMetropolisHastings` is equal to 0 -- i.e. if the DE-MC algorithm is always used instead of also having normal metropolis steps -- then the sampler attempts the operation `generation % crossover_gap` which ends up being `generation % 0` and you get a divide by zero error.

**Comment by bangerth** [comment]
I *think* that the variable is misnamed -- it's the interval, not the gap. If so, then it needs to be a positive number, and we should check that with an assertion.

**Comment by dklong-csu** [comment]
Makes sense, and that aligns with the comment above that comparison. Having `crossover_gap=1` would result in "perform crossover every 1 iteration".

**Comment by bangerth** [comment]
We should change the name of the variable, as well as update the documentation. I'll get around to it eventually, but at least now you know what you should choose.

**Comment by bangerth** [comment]
I've also special cased the current behavior to avoid the division by zero, see #188 .
